mono/profiler/log.c

```diff
@@ -16,7 +16,18 @@
 #include <mono/metadata/class-internals.h>
 #include <mono/metadata/debug-helpers.h>
 #include <mono/metadata/loader.h>
+#if 0 //FIXME
 #include <mono/metadata/loader-internals.h>
+#else
+#ifdef __cplusplus
+template <typename T>
+inline void
+mono_add_internal_call (const char *name, T method)
```

Comment:
Just modify the `ADD_ICALL` macro to do the cast; no need for a template hack IMO.

Reply:
And we are certain that, on the targets where we care about runtime size, those templates don't result in a bunch of pointless instantiations increasing the size of the final library?

Reply:
Yes. It is a one line inline function, that casts and calls the original.

If we templatized “deeply”, and didn’t inline, most?all systems combine identical functions (Microsoft linker got this feature circa 1996 and it isn’t alone.)